samples/dnn/audio_visual_speech_recognition.py

```diff
@@ -0,0 +1,264 @@
+        return pred
+
+    def run(self):
+        '''
+        Read the video and process it.
+        '''
+        self.cap.open(self.source, cv.CAP_MSMF, self.params)
+        if not self.cap.isOpened():
+            print('Cannot open video source')
+            exit(1)
+        audioBaseIndex = int(self.cap.get(cv.CAP_PROP_AUDIO_BASE_INDEX))
+        audioChannels = int(self.cap.get(cv.CAP_PROP_AUDIO_TOTAL_CHANNELS))
+
+        if self.cap.isOpened():
+            while self.cap.grab():
+                ret, frame = self.cap.retrieve()
+                frame = cv.resize(frame, (self.width, self.height))
+                audioFrame = np.asarray([])
+                audioFrame = self.cap.retrieve(audioFrame, audioBaseIndex)
+                audioFrame = audioFrame[1][0] if audioFrame is not None else None
+
+                if self.realtime:
+                    img, aud = self.preprocess(frame, audioFrame)
+                    if img is not None and aud is not None:
+                        img = cv.resize(img, (96,96))
+                        img = cv.cvtColor(img, cv.COLOR_BGR2GRAY)
+                        self.frames_queue.append(img)
+                        for audio in aud:
+                            self.audio_queue.append(audio)
+                        if len(self.frames_queue) < self.margin:
+                            continue
+                        pred = self.predict(self.frames_queue, self.audio_queue)
+                        print(pred)
+                        if self.show_video:
+                            cv.circle(frame, np.mean(self.landmarks_queue, axis=0)[0][5].astype(np.int32), 2, (0,0,255), -1)
+                            cv.circle(frame, np.mean(self.landmarks_queue, axis=0)[0][6].astype(np.int32), 2, (0,0,255), -1)
+                            cv.imshow('Audio-visual speech recognition in OpenCV', frame)
+                            if cv.waitKey(1) & 0xFF == ord('q'):
+                                break
+                else:
+                    self.frames_queue.append(frame)
+                    self.audio_queue.append(audioFrame)
+        if not self.realtime:
+            image_queue = []
+            audio_queue = []
+            for i in range(len(self.frames_queue)):
+                img, aud = self.preprocess(self.frames_queue[i], self.audio_queue[i])
+                if img is not None and aud is not None:
+                    img = cv.resize(img, (96,96))
+                    img = cv.cvtColor(img, cv.COLOR_BGR2GRAY)
+                    image_queue.append(img)
+                    for audio in aud:
+                        audio_queue.append(audio)
+                    if len(image_queue) / self.margin > 1:
+                        pred = self.predict(image_queue, audio_queue)
+                        print(pred)
+
+def parse_args():
+    parser = argparse.ArgumentParser(description='Audio Visual Speech Recognition')
+    backends = (cv.dnn.DNN_BACKEND_DEFAULT, cv.dnn.DNN_BACKEND_OPENCV)
```

Comment:
What about targets?

Reply:
What is the problem with other backends?

Reply:
What are the types of errors?

Reply:
Not implemented error

```
PS C:\Codes\GSoC\GSoC22\opencv\samples\dnn> python .\audio_visual_speech_recognition.py --show_video --input test.mp4 --target 0 --backend 2
Traceback (most recent call last):
  File "C:\Codes\GSoC\GSoC22\opencv\samples\dnn\audio_visual_speech_recognition.py", line 266, in <module>
    main()
  File "C:\Codes\GSoC\GSoC22\opencv\samples\dnn\audio_visual_speech_recognition.py", line 263, in main
    recognizer.run()
  File "C:\Codes\GSoC\GSoC22\opencv\samples\dnn\audio_visual_speech_recognition.py", line 213, in run
    pred = self.predict(image_queue, audio_queue)
  File "C:\Codes\GSoC\GSoC22\opencv\samples\dnn\audio_visual_speech_recognition.py", line 156, in predict
    out = self.model.forward()
cv2.error: OpenCV(4.6.0) D:\a\opencv-python\opencv-python\opencv\modules\dnn\src\legacy_backend.cpp:123: error: (-213:The function/feature is not implemented) Unknown backend identifier in function 'cv::dnn::dnn4_v20220524::detail::wrapMat'
```

Reply:
This means you don't build opencv with openvino or install opencl. You can run the classification sample with the same backends and targets to ensure that opencv has been build successfully.

Reply:
Yeah, I think openCL is not there in my local machine.

Reply:
Please install suitable drivers to your machine

Reply:
There's an issue. I'm using WSL on windows. It's not working on WSL. To install and build openCV on windows would take some time and since my mid terms are going on, I'm unable to do so. If possible, can you test the backends on your machine? 😅